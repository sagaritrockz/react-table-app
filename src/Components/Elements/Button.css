.btn {
    display: inline-block;
    font-weight: 400;
    text-align: center;
    white-space: nowrap;
    vertical-align: middle;
    -webkit-user-select: none;
    -moz-user-select: none;
    -ms-user-select: none;
    user-select: none;
    border: 1px solid transparent;
    padding: 5px 10px;
    font-size: 14px;
    line-height: 1.5;
    border-radius: 0.25rem;
    cursor: pointer;
    transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out,
        border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
}
.btn-custom {
    margin: 0 5px;
}