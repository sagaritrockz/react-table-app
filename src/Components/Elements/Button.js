import React from "react"
import { string, func } from 'prop-types'
import "./Button.css"

const Button = ({ label, click, name }) => {
    return (
        <button className="btn btn-custom" onClick={click} name={name}>
            {label}
        </button>
    );
};

Button.defaultProps = {
    label: "",
    click: f => f,
    name: ""
};

Button.propTypes = {
    label: string.isRequired,
    click: func.isRequired,
    name: string.isRequired
};

export default Button;
