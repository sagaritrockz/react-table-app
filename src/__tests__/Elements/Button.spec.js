import React from 'react'
import Enzyme from 'enzyme'
import Adaptar from 'enzyme-adapter-react-16.3'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import Button from '../../Components/Elements/Button'

Enzyme.configure({adapter: new Adaptar()});

describe('Button', () => {
    it('should render correctly', () => {
        const output = shallow(
          <Button click={function() {}} label="MockLabel" name="MockName" />
        );
        expect(shallowToJson(output)).toMatchSnapshot();
    });
});