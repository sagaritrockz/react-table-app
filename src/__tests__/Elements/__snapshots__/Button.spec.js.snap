// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly 1`] = `
<button
  className="btn btn-custom"
  name="MockName"
  onClick={[Function]}
>
  MockLabel
</button>
`;
